src/main/java/net/minestom/server/extensions/Extension.java

```diff
@@ -14,6 +19,14 @@
     @SuppressWarnings("unused")
     private Logger logger;
 
+    /**
+     * Observers that will be notified of events related to this extension.
+     * Kept as WeakReference because entities can be observers, but could become candidate to be garbage-collected while
+     * this extension holds a reference to it. A WeakReference makes sure this extension does not prevent the memory
+     * from being cleaned up.
+     */
+    private Set<WeakReference<IExtensionObserver>> observers = Collections.newSetFromMap(new ConcurrentHashMap<>());
```

Comment:
You probably want a way to "cleanup" the Set http://www.java2s.com/Code/Java/Collections-Data-Structure/Abagofweaklyreferencedobjects.htm

Reply:
Fix in e40186a2aad5eca86a8cc9dd408bc1629ff5d267